### PR TITLE
refactor: fix numerous compilation warnings

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
please see individual commits

### CODE REVIEW

File: src/wallet/wallet.cpp [hunk 1]
```diff
@@ -2773,8 +2773,8 @@ CAmount CWallet::GetAnonymizableBalance(bool fSkipDenominated, bool fSkipUnconfi
 {
     if (!CCoinJoinClientOptions::IsEnabled()) return 0;
 
-    std::vector<CompactTallyItem> vecTally;
-    if(!SelectCoinsGroupedByAddresses(vecTally, fSkipDenominated, true, fSkipUnconfirmed)) return 0;
+    std::vector<CompactTallyItem> vecTally = SelectCoinsGroupedByAddresses(fSkipDenominated, true, fSkipUnconfirmed);
+    if (vecTally.size() == 0) return 0;
```
Comment (UdjinM6) [inline src/wallet/wallet.cpp:2777]: empty()

File: src/coinjoin/client.cpp [hunk 2]
```diff
@@ -1332,7 +1332,7 @@ bool CCoinJoinClientSession::PrepareDenominate(int nMinRounds, int nMaxRounds, s
 
     // Try to add up to COINJOIN_ENTRY_MAX_SIZE of every needed denomination
     for (const auto& entry : vecTxDSIn) {
-        if (nSteps >= COINJOIN_ENTRY_MAX_SIZE) break;
+        if (nSteps >= int{COINJOIN_ENTRY_MAX_SIZE}) break;
```
Comment (UdjinM6) [inline src/coinjoin/client.cpp:1335]: could make `nSteps` `size_t` instead (same for other similar cases)

File: src/evo/cbtx.cpp [hunk 3]
```diff
@@ -215,15 +215,15 @@ bool CalcCbTxMerkleRootQuorums(const CBlock& block, const CBlockIndex* pindexPre
             auto qcHash = ::SerializeHash(qc.commitment);
             const auto& llmq_params = llmq::GetLLMQParams(qc.commitment.llmqType);
             auto& v = qcHashes[llmq_params.type];
-            if (v.size() == llmq_params.signingActiveQuorumCount) {
+            if (v.size() == uint64_t(llmq_params.signingActiveQuorumCount)) {
```
Comment (UdjinM6) [inline src/evo/cbtx.cpp:218]: `size_t(...)` maybe? or make `signingActiveQuorumCount` `size_t` (same for other `LLMQParams` members, not sure why we use `int` for them...)

File: src/prevector.h [hunk 4]
```diff
@@ -555,7 +555,7 @@ class prevector {
         // This avoids internal use of std::copy and operator++ on the iterators and instead allows efficient memcpy/memmove
         if (std::is_trivially_constructible<T>::value) {
             auto s = e - b;
-            if (v.size() != s) {
+            if (v.size() != uint64_t(s)) {
```
Comment (UdjinM6) [inline src/prevector.h:558]: why not `size_t`? (same for other similar cases below)

File: src/llmq/signing_shares.cpp [hunk 5]
```diff
@@ -811,7 +811,8 @@ void CSigSharesManager::TryRecoverSig(const CQuorumCPtr& quorum, const uint256&
 
 CDeterministicMNCPtr CSigSharesManager::SelectMemberForRecovery(const CQuorumCPtr& quorum, const uint256 &id, int attempt)
 {
-    assert(attempt < quorum->members.size());
+    assert(attempt >= 0);
```
Comment (UdjinM6) [inline src/llmq/signing_shares.cpp:814]: `attempt` could be `size_t` here and in `CSignedSession`

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Comment by PastaPastaPasta** [comment]
see latest

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**